lib/GraphQL/Execution.pm

```diff
@@ -623,6 +627,28 @@ fun _get_argument_values(
   \%coerced_values;
 }
 
+fun _coerce_for_error(Any $value) {
+#   my ($value) = @_;
+  
+  my $ret;
+  
+  if ( 'ARRAY' eq ref $value ) {
```

Comment:
This needs to be a recursive function, with a branch for `SCALAR`, and the `ARRAY` and `HASH` doing the obvious recurse rather than each having their `SCALAR` check.